myapp/app/views/tasks/index.html.erb

```diff
@@ -1,68 +1,98 @@
-        <th><%= t("task.table_headers.description") %></th>
-        <th><%= t("task.table_headers.priority") %></th>
-        <th><%= t("task.table_headers.deadline") %></th>
-        <th><%= t("task.table_headers.status") %></th>
-        <th><%= t("task.table_headers.labels") %></th>
-        <th><%= t("task.table_headers.created_at") %></th>
-        <th><%= t("task.table_headers.actions") %></th>
-      </tr>
-    </thead>
-    <tbody>
-      <% @tasks.each do |task| %>
-        <tr>
-          <td><%= task.name %></td>
-          <td><%= task.description %></td>
-          <td><%= task.priority %></td>
-          <td><%= task.deadline %></td>
-          <td><%= task.status %></td>
-          <td>
-            <% task.labels.each do |label| %>
-              <%= label.name %>
-            <% end %>
-          </td>
-          <td><%= task.created_at.strftime("%Y-%m-%d %H:%M:%S") %></td>
-          <td>
-            <%= link_to t('task.show_task_link'), task_path(task) %>
-            <%= link_to t('task.edit_task_link'), edit_task_path(task) %>
-            <%= link_to t('task.destroy_task_link'), task_path(task), 
-              method: :delete,
-              data: { confirm: t('task.confirm_delete_task') } %>
-          </td>
-        </tr>
+<div class="container">
+  <h1 ><%= t('page_titles.tasks_index') %></h1>
+  <% if @tasks.any? %>
+    <div class="search-form-container">
+      <%= form_tag tasks_path, method: :get, class: 'search-form' do %>
+        <div class="input-group">
+          <%= text_field_tag :name, params[:name], class: 'form-control', placeholder: "task name..." %>
+          <%= select_tag :status, options_for_select([
+            [t('task.status_placeholder'), ''],
+            [t('task.status_options.not_started'), 'Not Started'],
+            [t('task.status_options.in_progress'), 'In Progress'],
+            [t('task.status_options.done'), 'Done']], params[:status]),
+            class: 'form-control' %>
+          <div class="input-group-append">
+            <button class="btn btn-outline-secondary" type="submit">
+              <i class="fas fa-search"></i>
+            </button>
+          </div>
+        </div>
       <% end %>
-    </tbody>
-  </table>
-  
-  <%= paginate @tasks %>
-<% else %>
-  <p><%= t('task.no_tasks_found') %> <%= link_to 'Back to home', tasks_path %> </p>
-<% end %>
+    </div>
+
 
-<%= link_to t('task.new_task_link'), new_task_path %>
+    <div class="table-responsive">
+      <table class="table table-striped table-bordered">
+        <thead>
+          <tr>
+            <th><%= t("task.table_headers.name") %></th>
+            <th><%= t("task.table_headers.description") %></th>
+            <th><%= t("task.table_headers.priority") %></th>
+            <th>
+              <%= link_to t("task.table_headers.deadline"), tasks_path(sort_by: 'deadline', sort_order: params[:sort_by] == 'deadline' && params[:sort_order] == 'asc' ? 'desc' : 'asc') %>
```

Comment:
[nits]
Since there are several similar codes, why don't you implement them all together in a helper class?
```
// link_to
link_to t("task.table_headers.deadline"), tasks_path(sort_by: 'deadline', sort_order: link_sort_order(params, 'deadline'))

// fas fa-caret-***
<% if sort_order(params, 'deadline') == 'asc' %>
  <i class="fas fa-caret-up"></i>
<% elsif sort_order(params, 'deadline') == 'desc' %>
  <i class="fas fa-caret-down"></i>
<% end %>

// helper method
def link_sort_order(params, sort_by)
  sort_order(params, sort_by) == 'asc' ? 'desc' : 'asc'
end

def sort_order(params, sort_by)
  return nil unless params[:sort_by] == sort_by 
  params[:sort_order] == 'asc' ? 'desc' : 'asc'
end
```

Reply:
Done. Thanks for your advise, by adding helper method can help improve the maintainability and readability of my code.